sklearn/preprocessing/tests/test_data.py

```diff
@@ -968,6 +968,40 @@ def test_quantile_transform_check_error():
                          transformer.transform, 10)
 
 
+@pytest.mark.parametrize(
+    "missing_value, dtype",
+    [(np.nan, np.float64),
+     (np.nan, np.float32)])
+def test_quantile_transform_missing_value(missing_value, dtype):
+    X_some_missing = np.array([[0, 1],
+                               [0, 0],
+                               [missing_value, 2],
+                               [0, missing_value],
+                               [0, 1]], dtype=dtype)
+    X_all_missing = np.array([[missing_value, missing_value],
+                              [missing_value, missing_value]], dtype=dtype)
+    X_expected_some_missing = np.array([[0, 0.5],
```

Comment:
How about we establish a common test for column-wise preprocessors along the lines of:

```py
def test_missing_value_handling(est, X, n_missing):
    X[randint(X.shape[0], size=n_missing), randint(X.shape[1], size=n_missing)] = np.nan
    X_train, X_test = train_test_split(X)
    Xt = est.fit(X_train).transform(X_test)
    # missing values should still be missing, and only them
    assert_array_equal(np.isnan(Xt), np.isnan(X_test))

    for i in range(X.shape[1]):
        # train only on non-NaN
        est.fit(X_train[:, i][~np.isnan(X_train[:, i])])
        # check transforming with NaN works even when training without NaN
        Xt_col = est.transform(X_test[:, i])
        assert_array_equal(Xt_col, Xt[:, i])
        # check non-NaN is handled as before
        Xt_col_nonan = est.transform(X_test[:, i][~np.isnan(X_test[:, i])])
        assert_array_equal(Xt_col_nonan, Xt_col[~np.isnan(Xt_col)])
```

Reply:
I note that this does not test the all-NaN column in training case, which we need to work out how to handle.

Reply:
Where do you want to put this common test.
I thought to put it inside the `estimator_checks` and add a new `_yield_preprocessor_checks` but we don't have any tag to find out those. So we would require to make a list of it. Does it seem reasonable or @jnothman did you have something else in mind?